misc/python/materialize/checks/all_checks/sink.py

```diff
@@ -976,6 +978,61 @@ def validate(self) -> Testdrive:
         )
 
 
+@externally_idempotent(False)
+class AlterSinkMv(Check):
+    """Check ALTER SINK with materialized views"""
+
+    def _can_run(self, e: Executor) -> bool:
+        return self.base_version > MzVersion.parse_mz("v0.134.0-dev")
+
+    def initialize(self) -> Testdrive:
+        return Testdrive(
+            schemas()
+            + dedent(
+                """
+                > CREATE MATERIALIZED VIEW mv_alter1 AS (SELECT 1 AS a)
+                > CREATE SINK sink_alter_mv FROM mv_alter1
+                  INTO KAFKA CONNECTION kafka_conn (TOPIC 'sink-alter-mv')
+                  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+                  ENVELOPE DEBEZIUM
+                """
+            )
+        )
+
+    def manipulate(self) -> list[Testdrive]:
+        return [
+            Testdrive(dedent(s))
+            for s in [
+                """
+                > CREATE MATERIALIZED VIEW mv_alter2 AS (SELECT 2 AS a)
+                > ALTER SINK sink_alter_mv SET FROM mv_alter2;
+                """,
+                """
+                > CREATE MATERIALIZED VIEW mv_alter3 AS (SELECT 3 AS a)
+                > ALTER SINK sink_alter_mv SET FROM mv_alter3;
+                """,
+            ]
+        ]
+
+    def validate(self) -> Testdrive:
+        return Testdrive(
+            dedent(
+                """
+                > CREATE SOURCE sink_alter_mv_source_src
+                  FROM KAFKA CONNECTION kafka_conn (TOPIC 'sink-alter-mv')
+                > CREATE TABLE sink_alter_mv_source FROM SOURCE sink_alter_mv_source_src (REFERENCE "sink-alter-mv")
+                  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+                  ENVELOPE NONE
+
+                > SELECT before IS NULL, (after).a FROM sink_alter_mv_source
+                true 3
```

Comment:
@def- - I don't think this is true in general... if the sink publishes its snapshot I don't think we'll ever retract the old value and replace it with the new one. ([ALTER SINK has awkward semantics in this way](https://materialize.com/docs/sql/alter-sink/#cutover-scenarios).)

Reply:
Oh, you are right. My bad, will fix it.

Reply:
Ah, just pushed a change for this -- but not one I'm attached to if you have other ideas!

Reply:
My idea is a sleep, and checking that we actually get values from all 3 MVs. I'll push it in a minute

Reply:
Cool, yeah, I suspect borrowing the approach from `AlterSink` should work fine here too!